src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -353,6 +353,16 @@ public bool Equals(NuGetFramework other)
             return Comparer.Equals(this, other);
         }
 
+        public static bool operator ==(NuGetFramework left, NuGetFramework right)
+        {
+            return Comparer.Equals(left, right);
+        }
+
+        public static bool operator !=(NuGetFramework left, NuGetFramework right)
+        {
+            return !(left == right);
```

Comment:
I basically copied what `FrameworkVersion` does - https://referencesource.microsoft.com/#System/sys/system/runtime/versioning/FrameworkName.cs,218. I guess it makes the equality conditions more localized. I'm open to change this if you feel strongly about it.